installer.py

```
@@ -0,0 +1,110 @@
+from urllib.request import urlretrieve
+from zipfile import ZipFile
+
+class Installer:
+    base_link = "http://apertium.projectjj.com/{}{}"
+
+    def __init__(self, languages):  # type: (Installer, list) -> None
+        self._install_path = os.getenv("LOCALAPPDATA")
+        self._apertium_path = os.path.join(self._install_path, "apertium-all-dev")
+        self._download_path = tempfile.mkdtemp()
+        self._languages = languages
+        logging.basicConfig(filename="installer.log", format="%(asctime)s %(message)s",
+                            filemode="w", level=logging.DEBUG)
+        self._logger = logging.getLogger()
+        self._logger.setLevel(logging.DEBUG)
+
+    def _download_zip(self, download_files, extract_path):  # type: (Installer, dict, str) -> None
+
+        for zip_name, zip_link in download_files.items():
+            zip_download_path = os.path.join(self._download_path, zip_name)
+            urlretrieve(zip_link, filename=zip_download_path)
+            self._logger.info("%s download completed", zip_name)
+
+            # Extract the zip
+            zip_file = ZipFile(zip_download_path)
+            zip_file.extractall(path=extract_path)
+            zip_file.close()
+            self._logger.info("%s Extraction completed", zip_name)
+            os.remove(zip_download_path)
+            self._logger.info("%s removed", zip_name)
+
+    def _download_apertium_windows(self):  # type: (Installer) -> None
+        """Installs Apertium-all-dev to %localappdata%"""
+
+        apertium_windows = {
+            "apertium-all-dev.zip":
+                Installer.base_link.format("/win64/nightly/apertium-all-dev.zip", "")
+        }
+
+        self._download_zip(apertium_windows, self._install_path)
+
+    def _download_package(self):  # type: (Installer) -> None
+        """Installs Language Data to Apertium"""
+
+        zip_path = ""
+        if platform.system() == "Windows":
+            zip_path = "win32/nightly/data.php?zip="
```

Let's raise an error in the `else` clause. That will eliminate the assignment above.

---

What should be the error message in the else part?

---

`Platform {platform.system()} package install not supported` or something?